dependencies.yaml

```diff
@@ -117,13 +144,22 @@ dependencies:
               - python=3.10
   run:
     common:
-      - output_types: [conda, requirements]
+      - output_types: [conda, requirements, pyproject]
         packages:
           - numba>=0.49
           - numpy>=1.19
+  cuda-python:
+    common:
+      - output_types: [conda, requirements, pyproject]
+        packages:
+          - cuda-python>=11.7.1,<12.0
   test_python:
     common:
-      - output_types: [conda, requirements]
+      - output_types: [conda, requirements, pyproject]
         packages:
           - pytest
+  test_python_cov:
```

Comment:
Why is this split out?

Reply:
Because pytest-cov isn't an actual testing requirement for rmm, it is just what we use in our CI. I don't want it included in pyproject.toml (or eventually meta.yaml) where it will become an actual dependency for installing the package.

Reply:
Both pytest and pytest-cov would seem appropriate as “test extras” to me.

Reply:
Went with this change. All pytest plugins are now listed as test dependencies.